iothub/service/src/Registry2/ConfigurationsClient.cs

```diff
@@ -0,0 +1,460 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Common.Exceptions;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Subclient of <see cref="IotHubServiceClient"/> that handles creating, updating, getting and deleting configurations
+    /// </summary>
+    public class ConfigurationsClient
+    {
+        private const string ConfigurationRequestUriFormat = "/configurations/{0}?{1}";
+        private const string ConfigurationsRequestUriFormat = "/configurations/?top={0}&{1}";
+        private const string ApplyConfigurationOnDeviceUriFormat = "/devices/{0}/applyConfigurationContent?" + ClientApiVersionHelper.ApiVersionQueryString;
+        private static readonly TimeSpan s_defaultOperationTimeout = TimeSpan.FromSeconds(100);
+
+        private IHttpClientHelper _httpClientHelper;
+
+        /// <summary>
+        /// Creates an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        protected ConfigurationsClient()
+        {
+        }
+
+        internal ConfigurationsClient(IotHubConnectionProperties connectionProperties, HttpTransportSettings transportSettings)
+        {
+            _httpClientHelper = new HttpClientHelper(
```

Comment:
This client is created for you at the serviceClient level and should be passed down. Each subclient does not need to create its own http client

Reply:
See [here](https://github.com/Azure/azure-iot-sdk-csharp/blob/fc7c312d6986d4426e935ab0feb45a62f03751d1/iothub/service/src/Registry2/DevicesClient.cs#L44)